### PR TITLE
added notation CLI reference docs based on RC4

### LINKED ISSUE #163: Create a chapter for CLI reference doc
We can generate the CLI reference doc for Notation CLI using [Cobra](https://umarcor.github.io/cobra/#documentation-generation). I think we can add the CLI reference doc to the Notary documentation, which is similar to https://oras.land/cli_reference/. It will provide fundamental references for users based on the CLI help information.

### PR DESCRIPTION
**Pull request by zr-msft**
fixes #163

### CODE REVIEW

**Review by yizha1 — APPROVED** [review]
LGTM. 
@zr-msft One question: I didn't see sub-commands for `notation sign` and `notation verify`. Do you plan to add them later?

**Comment by FeynmanZhou** [comment]
@zr-msft Could you add a code block effect for each example like https://oras.land/cli_reference/? It would be much more convenient for users to copy.  We will also add a **Copy** button to the code block soon. The layout will be much more comfortable if we put the example CLI command in the code block.

**Comment by FeynmanZhou** [comment]
@yizha1  There are reference docs for `notation sign` and `notation verify`:

https://deploy-preview-183--notarydev.netlify.app/docs/cli-reference/notation_sign/
https://deploy-preview-183--notarydev.netlify.app/docs/cli-reference/notation_verify/

File: content/en/docs/cli-reference/notation.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: notation
+---
+Notation - a tool to sign and verify artifacts
+
```
Comment (FeynmanZhou) [inline content/en/docs/cli-reference/notation.md:5]: Should we keep this file? It doesn't have substantial content.

**Comment by zr-msft** [comment]
manually added code blocks to examples for now. Still need to research better automated generation of these changes, but thats out of scope for this PR

**Review by FeynmanZhou — APPROVED** [review]
LGTM

File: content/en/docs/cli-reference/notation.md (hunk 1 above)
Comment (yizha1) [reply content/en/docs/cli-reference/notation.md:5]: The output of `notation` and `notation -h` is the same.

**Review by yizha1 — APPROVED** [review]
LGTM for a first version

**Pull request merged**